src/workloads/query/TimeseriesExtendedRange.yml

```diff
@@ -0,0 +1,163 @@
+SchemaVersion: 2018-07-01
+Owner: Query Integration
+Description: |
+  This workload runs queries on time-series collections with data before the unix epoch (extended range).
+  Some optimizations can only be made on post-1970 data. This benchmark is intended to give us an idea
+  of how much performance we lose on extended range data.
+
+Keywords:
+  - timeseries
+  - aggregate
+
+GlobalDefaults:
+  Database: &database test
+  Collection: &collection Collection0
+  DocumentCount: &documentCount 1e7
+  Repeat: &repeat 200
```

Comment:
It looks like this benchmark only repeats the queries 200 times on a really small dataset. I think this is likely to lead to really noisy results that won't be representative of a real customer use case. 
- Have we done patch testing to validate that we are getting good signal here?
- If there was a regression in this workload, would it be high priority to fix?

Reply:
I could increase the number of queries run. Also, I didn't realize 10^7 is considered a small dataset. What is a realistic dataset size?

> Have we done patch testing to validate that we are getting good signal here?

Yea, the signal was not very noisy and also reflected the regression we are expecting to see after fixing https://github.com/10gen/mongo/pull/26614

> If there was a regression in this workload, would it be high priority to fix?

Yes, this is a benchmark we specifically optimize for, since it is a common pattern.

Reply:
Thanks Matt. How noisy was it? What does 10^7 documents map to in terms of GiB?

Reply:
I realized the benchmarks actually changed a bit since the start of this PR, so I'll remeasure the noise and get back to you.

The collection size is interesting. The initial size was small, so I added two more random double fields. With these fields, I think it's still affected by the fact that WiredTiger compresses data, and also time-series collections bucket similar data together. The size of the TS collection is 0.17 GB (using `db.collection.totalSize()`). The size of the equivalent non-TS collection is 0.52 GB.

Since both TS and non-TS collections are compressed, the uncompressed collection size (the data processed by the query engine) is probably in the GB

Reply:
I ran three patches with the latest benchmark, it looks not too noisy:

```
BlockProcessingExtendedRangeMinTime
ARM (AverageLatency, ns)
236,650,000
235,400,000
236,030,000
range = 1,250,000
range / avg measurement = 0.53%

Intel (AverageLatency, ns)
129,160,000
129,800,000
127,810,000
range = 1,990,000
range / avg measurement = 1.54%

BlockProcessingExtendedRangeMaxTime
ARM (AverageLatency, ns)
61,750,000
61,200,000
60,680,000
range = 1,070,000
range / avg measurement = 1.75%

Intel (AverageLatency, ns)
36,260,000
36,600,000
35,770,000
range = 830,000
range / avg measurement = 2.29%

```

Reply:
Thanks @mattBoros, 0.17GiB seems pretty small - is that how large these collections would be in practice? If we optimize for this smaller dataset, is there any risk we'd miss problems that occur on a larger collection?

Reply:
I could double the size, but I tried doing 10x more documents and insertion was taking a long time.

I think the uncompressed data size matters more than the compressed size for this benchmark, since the query system would be operating on the uncompressed data, which is probably at least a GB. I am curious if other benchmarks insert lots of data, like is it typical to have long insertion times? Then I'd be happy to increase it more